pkg/services/ngalert/state/historian/annotation.go

```diff
@@ -61,11 +61,19 @@ func (h *AnnotationBackend) Record(ctx context.Context, rule history_model.RuleM
 	annotations := buildAnnotations(rule, states, logger)
 	panel := parsePanelKey(rule, logger)
 
+	// New background job, so start with a completely new context to avoid pollution.
```

Comment:
I debated quite heavily on where to split this new context:
1. in `state/manager.go`, at historian call sites
2. in the implementations themselves, like we've done here.

I settled on this approach since the contexts are bound to the concurrency at the end of the day - especially so with the `cancel()`. Therefore the context split should follow the concurrency wherever it lives.

Long term, we want to "lift" the concurrency up to call sites of `Record()` (once we solve issues with pointers to states that might change underneath us). Same principle applies - when the concurrency moves to `manager`, so will the contexts - in the future.

Reply:
I think we should copy trace ID from the original context.

Reply:
Doing this now is **extremely** complex and warrants a separate PR. Grafana supports multiple tracing implementations and the way they deal with contexts is different, so our copying needs to be aware of all implementations. Unifying this is an extremely nontrivial extension to the `tracing` package.

Reply:
Eh, did it anyway, didn't end up being too bad